### PR TITLE
[DOP-20049] add PATCH 'v1/locations/{location_id}' endpoint

### PR DESCRIPTION
**Pull request by TiGrib**
## Change Summary

Add PATCH `/v1/locations/{location_id}` endpoint. Which add/update external_id for location.
In case if there is no sush location in db return Not Found.
Move errors and exceptions from server to commons folder to resolve circular imports.

## Related issue number

[DOP-20049]

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/data-rentgen/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: tests/test_server/test_locations/test_patch_locations.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    assert response.status_code == HTTPStatus.OK, response.json()
+    assert response.json() == {
+        "name": location.name,
+        "type": location.type,
+        "addresses": [{"url": address.url} for address in location.addresses],
+        "external_id": "external_id",
+    }
+
+
+async def test_update_location_external_id(
```
Comment (dolfinus) [inline tests/test_server/test_locations/test_patch_locations.py:43]: Please add test for writing null to external_id

File: data_rentgen/commons/errors/__init__.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # SPDX-FileCopyrightText: 2024 MTS PJSC
 # SPDX-License-Identifier: Apache-2.0
 
-from data_rentgen.server.errors.base import BaseErrorSchema
-from data_rentgen.server.errors.registration import (
+from data_rentgen.commons.errors.base import BaseErrorSchema
```
[the remark this answers is not in the document]
Comment (TiGrib) [reply data_rentgen/commons/errors/__init__.py:4]: There is a 'circular import' error in `db/repositories/location.py`.
And imo repositories shoud't know about `server`.
I can move logic of `update_external_id()` to route level and separate it into get and update methods.

[the remark this answers is not in the document]
Comment (dolfinus) [reply data_rentgen/commons/errors/__init__.py:4]: We,, exceptions can be moved to `data_rentgen.exceptions`. Schemas for now are server-only thing, and should stay in `data_rentgen.server` module

File: data_rentgen/server/errors/schemas/not_found.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+@register_error_response(
+    exception=EntityNotFoundError,
+    status=http.HTTPStatus.NOT_FOUND,
+)
+class NotFoundSchema(BaseErrorSchema):
+    code: Literal["not_found"] = "not_found"
+    details: NotFoundDetailsSchema
+
+    def to_exception(self) -> EntityNotFoundError:
```
Comment (dolfinus) [inline data_rentgen/server/errors/schemas/not_found.py:28]: Actually, this is used only in Horizon client, to convert HTTP response to local exception object. This can be removed.

**Review by dolfinus — APPROVED** [review]
